pkg/env/env_test.go

```diff
@@ -417,6 +416,49 @@ func TestEnv_Test(t *testing.T) {
 				return val
 			},
 		},
+		{
+			name: "before-and-after features unable to mutate feature",
+			ctx:  context.TODO(),
+			expected: []string{
+				"before-each-feature",
+				"test-feat-1",
+				"after-each-feature",
+				"before-each-feature",
+				"test-feat-2",
+				"after-each-feature",
+			},
+			setup: func(t *testing.T, ctx context.Context) []string {
+				env := newTestEnv()
+				val := []string{}
+				env.BeforeEachFeature(func(ctx context.Context, _ *envconf.Config, info features.Feature) (context.Context, error) {
+					val = append(val, "before-each-feature")
```

Comment:
We just gotta make sure test developers are not able to have access and call  `info.Steps()[0](...)` step functions. If they can't, ignore this.

Reply:
Added an extra check or two in this test to ensure values were/werent nil as expected.